src/main/java/appeng/fluids/parts/PartFluidStorageBus.java

```diff
@@ -0,0 +1,508 @@
+		this.getConfigManager().registerSetting( Settings.STORAGE_FILTER, StorageFilter.EXTRACTABLE_ONLY );
+		this.source = new MachineSource( this );
+	}
+
+	private IMEInventory<IAEFluidStack> getInventoryWrapper( TileEntity target )
+	{
+		EnumFacing targetSide = this.getSide().getFacing().getOpposite();
+		// Prioritize a handler to directly link to another ME network
+		IStorageMonitorableAccessor accessor = target.getCapability( Capabilities.STORAGE_MONITORABLE_ACCESSOR, targetSide );
+		if( accessor != null )
+		{
+			IStorageMonitorable inventory = accessor.getInventory( this.source );
+			if( inventory != null )
+			{
+				return inventory.getInventory( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+			}
+
+			// So this could / can be a design decision. If the tile does support our custom capability,
+			// but it does not return an inventory for the action source, we do NOT fall back to using
+			// IItemHandler's, as that might circumvent the security setings, and might also cause
+			// performance issues.
+			return null;
+		}
+
+		// Check via cap for IItemHandler
+		IFluidHandler handlerExt = target.getCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, targetSide );
+		if( handlerExt != null )
+		{
+			return new FluidHandlerAdapter( handlerExt, this );
+		}
+
+		return null;
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( IGridNode node, int TicksSinceLastCall )
+	{
+		if( this.resetCacheLogic != 0 )
+		{
+			this.resetCache();
+		}
+
+		if( this.monitor != null )
+		{
+			return this.monitor.onTick();
+		}
+
+		return TickRateModulation.SLEEP;
+	}
+
+	protected void resetCache()
+	{
+		final boolean fullReset = this.resetCacheLogic == 2;
+		this.resetCacheLogic = 0;
+
+		final IMEInventory<IAEFluidStack> in = this.getInternalHandler();
+		IItemList<IAEFluidStack> before = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+		if( in != null )
+		{
+			before = in.getAvailableItems( before );
+		}
+
+		this.cached = false;
+		if( fullReset )
+		{
+			this.handlerHash = 0;
+		}
+
+		final IMEInventory<IAEFluidStack> out = this.getInternalHandler();
+
+		if( in != out )
+		{
+			IItemList<IAEFluidStack> after = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+			if( out != null )
+			{
+				after = out.getAvailableItems( after );
+			}
+			Platform.postListChanges( before, after, this, this.source );
+		}
+	}
+
+	protected void resetCache( final boolean fullReset )
+	{
+		if( this.getHost() == null || this.getHost().getTile() == null || this.getHost().getTile().getWorld() == null || this.getHost().getTile().getWorld().isRemote )
+		{
+			return;
+		}
+
+		if( fullReset )
+		{
+			this.resetCacheLogic = 2;
+		}
+		else
+		{
+			this.resetCacheLogic = 1;
+		}
+
+		try
+		{
+			this.getProxy().getTick().alertDevice( this.getProxy().getNode() );
+		}
+		catch( final GridAccessException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)